core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedResultCollector.java

```diff
@@ -141,22 +141,52 @@ public DictionaryBasedResultCollector(BlockExecutionInfo blockExecutionInfos) {
       }
       fillMeasureData(scannedResult, row);
       if (scannedResult.complexParentIndexToQueryMap.toString().contains("StructQueryType")) {
+        int[] isComplexColumn = new int[queryDimensions.length + queryMeasures.length];
+        for (ProjectionDimension dimension : queryDimensions) {
+          if (null != dimension.getDimension().getComplexParentDimension()) {
+            isComplexColumn[dimension.getOrdinal()] = 1;
```

Comment:
only checking parent complex dimension. so can we name it isComplexChildColumn ?

Reply:
yes. If it has complex parent dimension it is a complex child column